### PR TITLE
Refactor for azure

### PR DESCRIPTION
**Pull request by pauldotyu**
## Purpose

This branch introduces significant changes to the order-service and makeline-service projects so that RabbitMQ and MongoDB can be swapped out for Azure Service Bus and Azure CosmosDB respectively. 

Since Azure Service Bus uses AMQP 1.0, the underlying libs for order-service and makeline-service has been swapped out with with ones that support this protocol. Also, local instances of RabbitMQ will have the amqp1_0 plugin enabled.

This will resolve issues #42 and #43.

## Does this introduce a breaking change?

```
[x] Yes, but only environment variable names
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone https://github.com/azure-samples/aks-store-demo
cd aks-store-demo
git checkout refactor-for-azure
```

* Run the code using docker compose

```
docker compose up --build
```

* You can also run the code using KIND

```
make all
```

* Test the code

You should see docker compose logs that shows orders being submitted by virtual-customer and processed by the virtual-worker.

Also, using a web browser, navigate to the store-front and store-admin sites.

> NOTE: If using a KIND cluster, you will need to port-forward the store-front and store-admin services using the command below.

```
kubectl port-forward svc/store-front 8080:80
kubectl port-forward svc/store-admin 8081:80
```

## What to Check
Verify that the following are valid
* Ensure the store-front can submit orders and orders can be processed from store-admin
* Ensure the ai-service generates product descriptions from OpenAI

### CODE REVIEW

File: src/makeline-service/README.md [hunk 1]
```diff
@@ -4,34 +4,125 @@ This is a Golang app that provides an API for processing orders. It is meant to
...
+```bash
+az group create --name <resource-group-name> --location <location>
+az servicebus namespace create --name <namespace-name> --resource-group <resource-group-name>
+az servicebus queue create --name orders --namespace-name <namespace-name> --resource-group <resource-group-name>
+```
+
+Once you have created the Service Bus namespace and queue, you will need to create a shared access policy with the **Listen** permission for the namespace.
+
+```bash
+az servicebus namespace authorization-rule create --name listener --namespace-name <namespace-name> --resource-group <resource-group-name> --queue-name orders --rights Listen
```
Comment (chzbrgr71) [inline src/makeline-service/README.md:46]: I think `--queue-name` is here by mistake. I removed it and the command worked.

File: src/makeline-service/README.md [hunk 2]
```diff
@@ -4,34 +4,125 @@ This is a Golang app that provides an API for processing orders. It is meant to
...
 ```
 
-With the services running, open a new terminal and navigate to the `makeline-service` directory. Then run the following commands:
+Next, get the connection information for the Azure Service Bus queue and save the values to environment variables.
 
 ```bash
-export ORDER_QUEUE_CONNECTION_STRING=amqp://username:password@localhost:5672/
+HOSTNAME=$(az servicebus namespace show --name <namespace-name> --resource-group <resource-group-name> --query serviceBusEndpoint -o tsv | sed 's/https:\/\///;s/:443\///')
+
+PASSWORD=$(az servicebus namespace authorization-rule keys list --namespace-name <namespace-name> --resource-group <resource-group-name> --queue-name orders --name listener --query primaryKey -o tsv)
```
Comment (chzbrgr71) [inline src/makeline-service/README.md:54]: I think --queue-name is here by mistake. I removed it and the command worked.

```
PASSWORD=$(az servicebus namespace authorization-rule keys list --namespace-name briarsbqueue --resource-group myResourceGroup --name listener --query primaryKey -o tsv)
```

Comment (chzbrgr71) [reply src/makeline-service/README.md:54]: On the order-service, you used `az servicebus queue authorization-rule keys list` but here it is on the namespace, so queue name is not needed

File: src/makeline-service/README.md [hunk 3]
```diff
@@ -4,34 +4,125 @@ This is a Golang app that provides an API for processing orders. It is meant to
...
+```
+
+Finally, set the environment variables.
+
+```bash
+export ORDER_QUEUE_URI=$HOSTNAME
```
Comment (chzbrgr71) [inline src/makeline-service/README.md:60]: This should be something like:

`export ORDER_QUEUE_URI=amqps://$HOSTNAME`

**Review by chzbrgr71 — COMMENTED** [review]
I put a couple comments below for the docs for makeline.

File: src/makeline-service/README.md (hunk 1 above)
Comment (pauldotyu) [reply src/makeline-service/README.md:46]: Removed the erroneous flag.

File: src/makeline-service/README.md (hunk 2 above)
Comment (pauldotyu) [reply src/makeline-service/README.md:54]: Removed the erroneous flag.

File: src/makeline-service/README.md (hunk 3 above)
Comment (pauldotyu) [reply src/makeline-service/README.md:60]: Updated the `ORDER_QUEUE_URI` to include `amqps://`

**Review by chzbrgr71 — APPROVED** [review]
Looks good now. I will merge

**Pull request merged**